app/controllers/pages_controller.rb

```diff
@@ -23,13 +20,17 @@ def update
       @page.update(page_params)
     end
 
-    redirect_to page_path
+    redirect_to page_path(path: @page.path)
   end
 
   private
 
   def set_page
     @page = Page.find_by(path: params[:path])
+    if @page.blank?
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.